tools/logger/convert.c

```diff
@@ -961,7 +964,8 @@ static int verify_ldc_checksum(const uint32_t ldc_sum)
 	count = fread(&ver, sizeof(ver), 1, global_config->version_fd);
 	if (!count) {
 		log_err("Error while reading %s.\n", global_config->version_file);
-		return -ferror(global_config->version_fd);
+		int result = -ferror(global_config->version_fd);
+		return result;
```

Comment:
Pointless refactor.

Reply:
This is not pointless refactor. Calling return on ferror causes a dangling pointer to global_config, which is no longer in scope. And static analysis tools reports that fact.

Reply:
This specific change would also have the exact same dangling pointer because it literally compiles to the same thing if you have at least -O1...

Reply:
This error is fatal, the tool exits when it happens. KW is not smart enough to see that, this is yet another false positive.

Indeed this change makes absolutely no difference, I hope KW is not so dumb to be deceived by it.